features/api_unattended_upgrades.feature

```diff
@@ -0,0 +1,182 @@
+        Then I will see the following on stdout:
+        """
+        0
+        """
+        When I run shell command `pro api u.pro.unattended_upgrades.status.v1 | jq .data.attributes.package_lists_refresh_frequency_days` as non-root
+        Then I will see the following on stdout:
+        """
+        1
+        """
+        When I run shell command `pro api u.pro.unattended_upgrades.status.v1 | jq .data.attributes.unattended_upgrades_running` as non-root
+        Then I will see the following on stdout:
+        """
+        false
+        """
+        When I run shell command `pro api u.pro.unattended_upgrades.status.v1 | jq .data.attributes.unattended_upgrades_disabled_reason` as non-root
+        Then I will see the following on stdout:
+        """
+        "APT::Periodic::Unattended-Upgrade is turned off"
+        """
+        When I run shell command `pro api u.pro.unattended_upgrades.status.v1 | jq '.data.meta.\"APT::Periodic::Unattended-Upgrade\"'` as non-root
+        Then I will see the following on stdout:
+        """
+        "0"
+        """
+        When I run `systemctl stop apt-daily.timer` with sudo
+        And I run `rm /etc/apt/apt.conf.d/99test` with sudo
+        And I run shell command `pro api u.pro.unattended_upgrades.status.v1 | jq .data.attributes.systemd_apt_timer_enabled` as non-root
+        Then I will see the following on stdout:
+        """
+        false
+        """
+        When I run shell command `pro api u.pro.unattended_upgrades.status.v1 | jq .data.attributes.unattended_upgrades_running` as non-root
+        Then I will see the following on stdout:
+        """
+        false
+        """
+        When I run shell command `pro api u.pro.unattended_upgrades.status.v1 | jq .data.attributes.unattended_upgrades_disabled_reason` as non-root
+        Then I will see the following on stdout:
+        """
+        "apt-daily.timer jobs are not running"
+        """
+        When I create the file `/etc/apt/apt.conf.d/50unattended-upgrades` with the following:
+        """
+        APT::Periodic::Unattended-Upgrade "1";
+        """
+        And I run `systemctl start apt-daily.timer` with sudo
+        And I run shell command `pro api u.pro.unattended_upgrades.status.v1 | jq .data.attributes.unattended_upgrades_frequency_days` as non-root
+        Then I will see the following on stdout:
+        """
+        1
+        """
+        When I run shell command `pro api u.pro.unattended_upgrades.status.v1 | jq .data.attributes.systemd_apt_timer_enabled` as non-root
+        Then I will see the following on stdout:
+        """
+        true
+        """
+        When I run shell command `pro api u.pro.unattended_upgrades.status.v1 | jq .data.attributes.unattended_upgrades_allowed_origins` as non-root
+        Then I will see the following on stdout:
+        """
+        []
```

Comment:
I'm surprised this is an empty list. Am I missing something in the test that sets this?

Reply:
Yes, this is empty because we modify `/etc/apt/apt.conf.d/50unattended-upgrades` which is the file that properly configures that variable

Reply:
Ahhhh I see now thanks!
I also just learned that the file in our repo `apt.conf.d/51ubuntu-advantage-esm` is not used I guess

Reply:
Good catch. I will create a follow up PR removing that file